test/integration/util/util.go

```diff
@@ -1157,6 +1157,8 @@ func NextPodOrDie(t *testing.T, testCtx *TestContext) *schedulerframework.Queued
 }
 
 // NextPod returns the next Pod in the scheduler queue, with a 5 seconds timeout.
+// Note that this function leaks goroutines in the case of timeout; even after this function returns after timeout,
+// the goroutine made by this function keep waiting to pop a pod from the queue.
```

Comment:
can you fix this in timeout()?

Reply:
If we want to fix it, I guess we have to modify schedulingQueue.Pop() to somehow abort waiting, I don't have any other idea to solve this leak.

Reply:
`Pop()` would be able to quit in another way I guess - we can use `defer schedulingQueue.Close()` in each test to signal the queue to quit _gracefully_, right?

Reply:
Each test, yes. 
Actually, we use a different scheduler in every test case, so this leaked goroutine won't be carried over to the next test case. 

What I meant here is that this NextPod is called multiple times in one test. What confused me when writing tests was:
1. The first call of `NextPod` is timeout-ed. 
2. Do something and Pod-A is moved to activeQ.
3. I want to confirm (2) moved Pod-A, and use `NextPod` again to make sure Pod-A is popped. But the goroutine made in (1) is still alive and pops Pod-A out. The second call of `NextPod` returns timeout.

Reply:
https://github.com/kubernetes/kubernetes/issues/123480